grpc/public/password_reset.go

```diff
@@ -0,0 +1,32 @@
+package public
+
+import (
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/app/services"
+	authnpb "github.com/keratin/authn-server/grpc"
+	context "golang.org/x/net/context"
+)
+
+var _ authnpb.PasswordResetServiceServer = passwordResetServer{}
+
+type passwordResetServer struct {
+	app *app.App
+}
+
+func (s passwordResetServer) RequestPasswordReset(ctx context.Context, req *authnpb.PasswordResetRequest) (*authnpb.PasswordResetResponse, error) {
+
+	account, err := s.app.AccountStore.FindByUsername(req.GetUsername())
+	if err != nil {
+		panic(err)
+	}
+
+	// run in the background so that a timing attack can't enumerate usernames
+	go func() {
+		err := services.PasswordResetSender(s.app.Config, account)
+		if err != nil {
+			s.app.Reporter.ReportError(err)
```

Comment:
should we now prefer `ReportGRPCError` in all grpc handlers?

* `grpc/private/unsecured.go`
* `grpc/public/password_reset.go`
* `grpc/public/passwordless.go`
* `grpc/public/public_server.go`

Reply:
Hmm... the problem is we don't have `*grpc.UnaryServerInfo` in here. This information is only available at the interceptor level. Should I pass it via `context`? Or, and this doesn't feel clean, panic and let the panic-recovery interceptor handle it which can inject `*grpc.UnaryServerInfo`?

Reply:
👌 i understand. i'm happy to defer this decision until we have real-world experience that motivates extra complexity.

Reply:
Actually, I've solved it locally but haven't pushed the changes yet. I opted to inject `*grpc.UnaryServerInfo` via interceptor through `context.Context`, and retrieve it from the context when needed.